poollet/machinepoollet/mem/mem.go

```diff
@@ -0,0 +1,18 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and IronCore contributors
+// SPDX-License-Identifier: Apache-2.0
+
+package mem
+
+import (
+	"context"
+	"errors"
+
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+var ErrNoMatchingMachineEvents = errors.New("no matching machine events")
+
+type MachineEventMapper interface {
+	manager.Runnable
+	WaitForSync(ctx context.Context) error
+}
```

Comment:
I think we can completely drop this interface and use  `manager.Runnable` directly. In the case of `MachineClasses` we can't proceed if we haven't received at least once classes from the provider therefore we have `WaitForSync`. In case of event's it's not needed.

Reply:
Removed interface and waitForSynch keeping it simple